Deprecation warnings doesn't seem to appear (v0.9.0-dev)

Hi there,

in previous versions (`0.8.x`) when marking a field as `Deprecated` an error like this would be shown to the user:

```bash
$ terraform plan
There are warnings and/or errors related to your configuration. Please
fix these before continuing.

Warnings:
  * data.test.test: "name_regex": [DEPRECATED] Use field 'name' instead

No errors found. Continuing with 1 warning(s).
```

In `v0.9.0-dev` this warning doesn't seem to appear, 
but you can see the validation error in the log when you run `TF_LOG=TRACE`:
```bash
2017/02/02 11:19:58 [ERROR] root: eval: *terraform.EvalValidateResource, err: Warnings: ["name_regex": [DEPRECATED] Use field 'name' instead]. Errors: []
```

Not exactly sure what this might be but hopefully this will help debugging the issue 🙂 

### Terraform Version
v0.9.0-dev

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. Mark a field as deprecated
2. `terraform plan` - No warning message appears
3. `TF_LOG=TRACE terraform plan 2>&1 | grep  "EvalValidateResource, err"` to see the warning

--- a/backend/local/backend_local.go
+++ b/backend/local/backend_local.go
@@ -1,6 +1,10 @@
 package local
 
 import (
+	"fmt"
+	"log"
+	"strings"
+
 	"github.com/hashicorp/errwrap"
 	"github.com/hashicorp/go-multierror"
 	"github.com/hashicorp/terraform/backend"
@@ -81,7 +85,23 @@ func (b *Local) context(op *backend.Operation) (*terraform.Context, state.State,
 		if b.OpValidation {
 			// We ignore warnings here on purpose. We expect users to be listening
 			// to the terraform.Hook called after a validation.
-			_, es := tfCtx.Validate()
+			ws, es := tfCtx.Validate()
+			if len(ws) > 0 {
+				// Log just in case the CLI isn't enabled
+				log.Printf("[WARN] backend/local: %d warnings: %v", len(ws), ws)
+
+				// If we have a CLI, output the warnings
+				if b.CLI != nil {
+					b.CLI.Warn(strings.TrimSpace(validateWarnHeader) + "\n")
+					for _, w := range ws {
+						b.CLI.Warn(fmt.Sprintf("  * %s", w))
+					}
+
+					// Make a newline before continuing
+					b.CLI.Output("")
+				}
+			}
+
 			if len(es) > 0 {
 				return nil, nil, multierror.Append(nil, es...)
 			}
@@ -90,3 +110,11 @@ func (b *Local) context(op *backend.Operation) (*terraform.Context, state.State,
 
 	return tfCtx, s, nil
 }
+
+const validateWarnHeader = `
+There are warnings related to your configuration. If no errors occurred,
+Terraform will continue despite these warnings. It is a good idea to resolve
+these warnings in the near future.
+
+Warnings:
+`

